packages/merge-styles/src/IStyleSet.ts

```diff
@@ -6,6 +6,8 @@ export type Diff<T extends keyof any, U extends keyof any> = ({ [P in T]: P } &
 
 export type Omit<U, K extends keyof U> = Pick<U, Diff<keyof U, K>>;
 
+export type TrimToFunction<T> = T extends (...args: any[]) => any ? T : never;
```

Comment:
I love this but we have intentionally been trying to avoid using "hard" TS 2.8 features, which is part of the reason why we aren't still yet on Typescript 3 and we have some unideal typings in here. 

Let me talk to the team to see if we can get this in.